client/azure.go

```diff
@@ -104,7 +104,7 @@ func NewAzureClientSet(config *AzureConfig) (*AzureClientSet, error) {
 // ResponseWasNotFound returns true if the response code from the Azure API
 // was a 404.
 func ResponseWasNotFound(resp autorest.Response) bool {
-	if resp.StatusCode == http.StatusNotFound {
+	if resp.Response != nil && resp.StatusCode == http.StatusNotFound {
```

Comment:
This is a simple bug fix I've sneaked in. The http.Response returned in autorest.Response can be nil depending on the error. So we need to check for it.